sonar-text-dotnet/pipeline_scripts/build_and_restore_solution.ps1

```diff
@@ -0,0 +1,26 @@
+$buildConfiguration="Release"
+
+cd $env:PROJECT_DIR
+
+Write-Host Restoring $env:SOLUTION_DIR
+nuget restore -LockedMode -Source "https://pkgs.dev.azure.com/sonarsource/399fb241-ecc7-4802-8697-dcdd01fbb832/_packaging/slvs_input/nuget/v3/index.json"  # nuget version is 6.3.1 https://github.com/SonarSource/re-ci-images/blob/master/ec2-images/build-base-windows-dotnet.pkr.hcl#L39
+
+$signArtifacts=$false
+if(($env:CIRRUS_BRANCH -eq "master" ) -or ($env:CIRRUS_BRANCH.StartsWith("branch-")))
+{
+    $signArtifacts=$true
+    Write-Host "signArtifacts variable was set to {$signArtifacts}".
+}
```

Comment:
Since this is not yet completed (and not used for real) I would remove the signing-related code from this PR. 

In this way we will be able the full signing logic in a dedicated PR, we will have the full context, and the PR will be easier to understand.